File: swagger-client/micronaut/src/main/java/io/swagger/model/MailStatsType.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.model.MailStatsTypeVolume;
import io.swagger.v3.oas.annotations.media.Schema;
import io.micronaut.core.annotation.Introspected;
import io.micronaut.validation.Validated;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Account usage statistics returned by &#x60;GET /mail/stats&#x60;.  Includes billing-cycle usage totals (for cost calculation) as well as time-windowed sent/received counts and volume breakdowns by IP, destination, and source address.
 */
@Schema(description = "Account usage statistics returned by `GET /mail/stats`.  Includes billing-cycle usage totals (for cost calculation) as well as time-windowed sent/received counts and volume breakdowns by IP, destination, and source address.")
@Validated
@Introspected

public class MailStatsType   {
  /**
   * The time window these `received`, `sent`, and `volume` statistics cover.
   */
  public enum TimeEnum {
    ALL("all"),
    BILLING("billing"),
    MONTH("month"),
    _7D("7d"),
    _24H("24h"),
    DAY("day"),
    _1H("1h");

    private String value;

    TimeEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static TimeEnum fromValue(String text) {
      for (TimeEnum b : TimeEnum.values()) {
        if (String.valueOf(b.value).equals(text)) {
          return b;
        }
      }
      return null;
    }
  }
  @JsonProperty("time")
  private TimeEnum time = TimeEnum._1H;

  @JsonProperty("usage")
  private Integer usage = null;

  @JsonProperty("currency")
  private String currency = null;

  @JsonProperty("cost")
  private Double cost = null;

  @JsonProperty("received")
  private Integer received = null;

  @JsonProperty("sent")
  private Integer sent = null;

  @JsonProperty("volume")
  private MailStatsTypeVolume volume = null;

  public MailStatsType time(TimeEnum time) {
    this.time = time;
    return this;
  }

  /**
   * The time window these `received`, `sent`, and `volume` statistics cover.
   * @return time
  **/
  @Schema(description = "The time window these `received`, `sent`, and `volume` statistics cover.")
  @NotNull

  public TimeEnum getTime() {
    return time;
  }

  public void setTime(TimeEnum time) {
    this.time = time;
  }

  public MailStatsType usage(Integer usage) {
    this.usage = usage;
    return this;
  }

  /**
   * Total messages accepted during the current billing cycle.  Used to calculate the `cost` value.
   * @return usage
  **/
  @Schema(description = "Total messages accepted during the current billing cycle.  Used to calculate the `cost` value.")
  @NotNull

  public Integer getUsage() {
    return usage;
  }

  public void setUsage(Integer usage) {
    this.usage = usage;
  }

  public MailStatsType currency(String currency) {
    this.currency = currency;
    return this;
  }

  /**
   * The ISO 4217 currency code for this account (e.g. `USD`).
   * @return currency
  **/
  @Schema(description = "The ISO 4217 currency code for this account (e.g. `USD`).")
  @NotNull

  public String getCurrency() {
    return currency;
  }

  public void setCurrency(String currency) {
    this.currency = currency;
  }

  public MailStatsType cost(Double cost) {
    this.cost = cost;
    return this;
  }

  /**
   * Estimated cost for the current billing cycle combining the base plan price and per-email charges ($0.20/1000 emails).
   * @return cost
  **/
  @Schema(description = "Estimated cost for the current billing cycle combining the base plan price and per-email charges ($0.20/1000 emails).")
  @NotNull

  public Double getCost() {
    return cost;
  }

  public void setCost(Double cost) {
    this.cost = cost;
  }

  public MailStatsType received(Integer received) {
    this.received = received;
    return this;
  }

  /**
   * Count of messages accepted by the relay within the selected `time` window. Includes messages still in queue.
   * @return received
  **/
  @Schema(description = "Count of messages accepted by the relay within the selected `time` window. Includes messages still in queue.")
  @NotNull

  public Integer getReceived() {
    return received;
  }

  public void setReceived(Integer received) {
    this.received = received;
  }

  public MailStatsType sent(Integer sent) {
    this.sent = sent;
    return this;
  }

  /**
   * Count of messages successfully delivered to the destination MX within the selected `time` window.  Will be ≤ `received`.
   * @return sent
  **/
  @Schema(description = "Count of messages successfully delivered to the destination MX within the selected `time` window.  Will be ≤ `received`.")
  @NotNull

  public Integer getSent() {
    return sent;
  }

  public void setSent(Integer sent) {
    this.sent = sent;
  }

  public MailStatsType volume(MailStatsTypeVolume volume) {
    this.volume = volume;
    return this;
  }

  /**
   * Get volume
   * @return volume
  **/
  @Schema(description = "")
  @NotNull

  @Valid
  public MailStatsTypeVolume getVolume() {
    return volume;
  }

  public void setVolume(MailStatsTypeVolume volume) {
    this.volume = volume;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    MailStatsType mailStatsType = (MailStatsType) o;
    return Objects.equals(this.time, mailStatsType.time) &&
        Objects.equals(this.usage, mailStatsType.usage) &&
        Objects.equals(this.currency, mailStatsType.currency) &&
        Objects.equals(this.cost, mailStatsType.cost) &&
        Objects.equals(this.received, mailStatsType.received) &&
        Objects.equals(this.sent, mailStatsType.sent) &&
        Objects.equals(this.volume, mailStatsType.volume);
  }

  @Override
  public int hashCode() {
    return Objects.hash(time, usage, currency, cost, received, sent, volume);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class MailStatsType {\n");
    
    sb.append("    time: ").append(toIndentedString(time)).append("\n");
    sb.append("    usage: ").append(toIndentedString(usage)).append("\n");
    sb.append("    currency: ").append(toIndentedString(currency)).append("\n");
    sb.append("    cost: ").append(toIndentedString(cost)).append("\n");
    sb.append("    received: ").append(toIndentedString(received)).append("\n");
    sb.append("    sent: ").append(toIndentedString(sent)).append("\n");
    sb.append("    volume: ").append(toIndentedString(volume)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

File: openapi-client/php-dt/src/App/DTO/MailStatsType.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\PhpAttribute as DTA;

/**
 * Account usage statistics returned by &#x60;GET /mail/stats&#x60;.  Includes billing-cycle usage totals (for cost calculation) as well as time-windowed sent/received counts and volume breakdowns by IP, destination, and source address.
 */
class MailStatsType
{
    /**
     * The time window these &#x60;received&#x60;, &#x60;sent&#x60;, and &#x60;volume&#x60; statistics cover.
     */
    #[DTA\Data(field: "time", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $time = null;

    /**
     * Total messages accepted during the current billing cycle.  Used to calculate the &#x60;cost&#x60; value.
     */
    #[DTA\Data(field: "usage", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $usage = null;

    /**
     * The ISO 4217 currency code for this account (e.g. &#x60;USD&#x60;).
     */
    #[DTA\Data(field: "currency", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "string"])]
    public string|null $currency = null;

    /**
     * Estimated cost for the current billing cycle combining the base plan price and per-email charges ($0.20/1000 emails).
     */
    #[DTA\Data(field: "cost", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "float"])]
    public float|null $cost = null;

    /**
     * Count of messages accepted by the relay within the selected &#x60;time&#x60; window. Includes messages still in queue.
     */
    #[DTA\Data(field: "received", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $received = null;

    /**
     * Count of messages successfully delivered to the destination MX within the selected &#x60;time&#x60; window.  Will be ≤ &#x60;received&#x60;.
     */
    #[DTA\Data(field: "sent", nullable: true)]
    #[DTA\Validator("Scalar", ["type" => "int"])]
    public int|null $sent = null;

    #[DTA\Data(field: "volume", nullable: true)]
    #[DTA\Strategy("Object", ["type" => \App\DTO\MailStatsVolume::class])]
    #[DTA\Validator("TypeCompliant", ["type" => \App\DTO\MailStatsVolume::class])]
    public \App\DTO\MailStatsVolume|null $volume = null;

}
